python/lsst/daf/persistence/policy.py

```diff
@@ -0,0 +1,252 @@
+            for k, v in u.iteritems():
+                if isinstance(d, collections.Mapping):
+                    if isinstance(v, collections.Mapping):
+                        r = doUpdate(d.get(k, {}), v)
+                        d[k] = r
+                    else:
+                        d[k] = u[k]
+                else:
+                    d = {k: u[k]}
+            return d
+        doUpdate(self.data, other)
+
+
+    #########################
+    # mimics pex_policy api #
+    #########################
+
+    def names(self, topLevelOnly=False):
+        """Get the dot-delimited name of all the keys in the hierarchy.
+        """
+        if topLevelOnly:
+            return self.keys()
+
+        def getKeys(d, keys, base):
+            for key in d.keys():
+                val = d[key]
+                levelKey = base + '.' + key if base is not None else key
+                keys.append(levelKey)
+                if isinstance(val, collections.Mapping):
+                    getKeys(val, keys, levelKey)
+        keys = []
+        getKeys(self.data, keys, None)
+        return keys
+
+    def getValue(self, name):
+        """Get the value for a parameter name/key. See class notes about dot-delimited access.
+
+        :param name:
+        :return: the value for the given name.
+        """
+        return self[name]
+
+    def setValue(self, name, value):
+        """Set the value for a parameter name/key. See class notes about dot-delimited access.
+
+        :param name:
+        :return: None
+        """
+        self[name] = value
+
+    def mergeDefaults(self, other):
+        """For any keys in other that are not present in self, sets that key and its value into self.
+
+        :param other: another Policy
+        :return: None
+        """
+        other.update(self)
+        self.data = other.data
+
+    def exists(self, key):
+        """Query if a key exists in this Policy
+
+        :param key:
+        :return: True if the key exists, else false.
+        """
+        return key in self
+
+    def getString(self, key):
+        """Get the string value of a key.
+
+        :param key:
+        :return: the value for key
+        """
+        return str(self[key])
+
+    def getBool(self, key):
+        """Get the value of a key.
+
+        :param key:
+        :return: the value for key
+        """
+        return bool(self[key])
+
+    def getPolicy(self, key):
+        """Get a subpolicy.
+
+        :param key:
+        :return:
+        """
+        return Policy(self[key])
+
+    def getStringArray(self, key):
+        """Get a value as an array. May contain one or more elements.
+
+        :param key:
+        :return:
+        """
+        val = self.get(key)
+        if isinstance(val, basestring):
+            val = [val]
+        elif not isinstance(val, collections.Container):
+            val = [val]
+        return val
+
+
+    ############
+    # yaml i/o #
+    ############
+
```

Comment:
If the inputs were updating an existing `Policy` I'd say they're OK, but as is they might better go into the constructor.  I'd say we should only support one dump format, so the `AsYaml` part of the name is redundant.  If I understand properly, pickling these should be trivial, correct?